esmvalcore/_recipe/check.py

```diff
@@ -456,3 +460,33 @@ def reference_for_bias_preproc(products):
             f"{len(reference_products):d}{ref_products_str}Please also "
             f"ensure that the reference dataset is not excluded with the "
             f"'exclude' option")
+
+
+def normalize_with_area_statistics(settings: dict) -> None:
+    """Check that normalize_with_area_statistics options are valid."""
+    if 'normalize_with_area_statistics' not in settings:
+        return
+    step_settings = settings['normalize_with_area_statistics']
+
+    # Check that operation is valid
+    operation = step_settings.get('operation', 'subtract')
+    valid_ops = ('subtract', 'divide')
+    if operation not in valid_ops:
+        raise RecipeError(
+            f"Invalid value encountered for `operation` in preprocessor "
+            f"`normalize_with_area_statistics`. Valid values are "
+            f"{valid_ops}. Got {operation}."
+        )
+
+    # Check that area_statistics_kwargs are valid
+    area_stats_kwargs = step_settings.get(
+        'area_statistics_kwargs', {'operator': 'mean'}
+    )
+    area_stats_settings_to_check = {'area_statistics': area_stats_kwargs}
+    try:
+        check_preprocessor_settings(area_stats_settings_to_check)
+    except (TypeError, ValueError) as exc:
+        raise RecipeError(
```

Comment:
agh - Codacy may be right this time - I am a bit uneasy to mask something that's not TypeError nor ValueError and let it through just as a user warning - that may trigger a fall off the cliff anyway but harder to catch where from

Reply:
it won't terminate it here if it doesn't catch one of the (TypeError, ValueError) - I mean it will, but down the line when things will get stuck somewhere else - if, say, `check_preprocessor_settings(area_stats_settings_to_check)` fails with OSError (for some bizarre reason), the OSError will show up as a warning in the stdout ie "Caught OSError etc etc" but the thing will keep on chugging. Stuff will hit the fan later though

Reply:
yes of course it will! I am a doofus, this is not in another try/except block of its own :man_facepalming: